ui/src/app/settings/components/GeneralForm/GeneralForm.js

```diff
@@ -0,0 +1,90 @@
+import { Formik } from "formik";
+import React from "react";
+import { useSelector, useDispatch } from "react-redux";
+import * as Yup from "yup";
+
+import actions from "app/settings/actions";
+import selectors from "app/settings/selectors";
+import Button from "app/base/components/Button";
+import Form from "app/base/components/Form";
+import Input from "app/base/components/Input";
+import Loader from "app/base/components/Loader";
+import Row from "app/base/components/Row";
+import Col from "app/base/components/Col";
+
+const GeneralSchema = Yup.object().shape({
+  maas_name: Yup.string().required(),
+  enable_analytics: Yup.boolean()
+});
+
+const GeneralForm = () => {
+  const dispatch = useDispatch();
+  const saveConfig = actions.config.save;
+
+  const maasName = useSelector(selectors.config.maasName);
+  const enableAnalytics = useSelector(selectors.config.enableAnalytics);
+
+  return (
+    <Row>
+      <Col size="6">
+        <Formik
+          initialValues={{
+            maas_name: maasName,
+            enable_analytics: enableAnalytics
+          }}
+          validationSchema={GeneralSchema}
+          onSubmit={(values, { setSubmitting }) => {
+            setTimeout(() => {
+              Object.keys(values).forEach(key => {
+                dispatch(saveConfig(key, values[key]));
+              });
+              setSubmitting(false);
+            }, 500);
+          }}
+        >
+          {({
+            errors,
+            isSubmitting,
+            touched,
+            handleSubmit,
+            handleChange,
+            handleBlur,
+            values
+          }) => (
+            <Form onSubmit={handleSubmit}>
+              <Input
```

Comment:
These can use the new Formik field component, see: https://github.com/canonical-web-and-design/maas-ui/pull/62#discussion_r312314179

Reply:
Done